File: src/main/java/com/deploy/bemyplan/service/auth/dto/request/LoginDto.java
package com.deploy.bemyplan.service.auth.dto.request;

import com.deploy.bemyplan.domain.user.UserSocialType;
import lombok.*;

import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;

@ToString
@Getter
@AllArgsConstructor(access = AccessLevel.PRIVATE)
@NoArgsConstructor(access = AccessLevel.PRIVATE)
public class LoginDto {

    @NotBlank(message = "{auth.token.notBlank}")
    private String token;

    @NotNull(message = "{user.socialType.notNull}")
    private UserSocialType socialType;

    public static LoginDto of(String token, UserSocialType socialType) {
        return new LoginDto(token, socialType);
    }
}

File: applications/app-bemyplan/src/main/kotlin/com/bemyplan/auth/application/LoginDto.kt
package com.bemyplan.auth.application

import com.deploy.bemyplan.domain.user.UserSocialType
import javax.validation.constraints.NotBlank
import javax.validation.constraints.NotNull

data class LoginDto(
    @NotBlank(message = "{auth.token.notBlank}")
    val token: String,
    @NotNull(message = "{user.socialType.notNull}")
    val socialType: UserSocialType
){
    companion object {
        fun of(token: String, socialType: UserSocialType): LoginDto {
            return LoginDto(token, socialType)
        }
    }
}
